commit 85636d1f21b0644e7dd47e3931479860c55b7a05
Author: Lekensteyn <lekensteyn@gmail.com>
Date:   Tue Feb 28 16:02:32 2012 +0100

    Remove switcheroo driver restriction
    
    Hacks (presumably) exist that make use of the switcheroo framework and enable
    card switching with the nvidia driver (gmux for mac?)

diff --git a/conf/bumblebee.conf.in b/conf/bumblebee.conf.in
index 783e8e3..e9414f2 100644
--- a/conf/bumblebee.conf.in
+++ b/conf/bumblebee.conf.in
@@ -37,7 +37,7 @@ AllowFallbackToIGC=@CONF_FALLBACKSTART@
 # PMMethod: method to use for saving power by disabling the nvidia card, valid
 # values are: auto - automatically detect which PM method to use
 #         bbswitch - new in BB 3, recommended if available
-#       switcheroo - vga_switcheroo method
+#       switcheroo - vga_switcheroo method, use at your own risk
 #             none - disable PM completely
 # https://github.com/Bumblebee-Project/Bumblebee/wiki/Comparison-of-PM-methods
 
diff --git a/src/bbconfig.c b/src/bbconfig.c
index d1ecefb..da7503b 100644
--- a/src/bbconfig.c
+++ b/src/bbconfig.c
@@ -175,8 +175,9 @@ void print_usage(int exit_val) {
                             none. auto selects a sensible method,\n\
                             bbswitch (kernel module) is available for nvidia\n\
                             and nouveau drivers,\n\
-                            switcheroo (vga_switcheroo) is for nouveau and\n\
-                            radeon drivers and none disables PM completely\n",
+                            switcheroo (vga_switcheroo) is usually for\n\
+			    nouveau and radeon drivers and none disables PM\n\
+			    completely\n",
             out);
 #ifdef WITH_PIDFILE
     fputs("\
diff --git a/src/switch/sw_switcheroo.c b/src/switch/sw_switcheroo.c
index b46cb9f..b71fe0d 100644
--- a/src/switch/sw_switcheroo.c
+++ b/src/switch/sw_switcheroo.c
@@ -87,12 +87,6 @@ int switcheroo_is_available(struct switch_info info) {
             " explicitly selected in the configuration.\n");
     return 0;
   }
-  if (strcmp("nouveau", info.driver)) {
-    /* switcheroo cannot be used with drivers other than nouveau */
-    bb_log(LOG_WARNING, "vga_switcheroo can only be used with the nouveau"
-            " driver, skipping method.\n");
-    return 0;
-  }
   return 1;
 }
 
